vault/src/system.rs

```diff
@@ -596,18 +595,21 @@ impl VaultService {
             .into_iter()
             .map(|(symbol, amount)| Ok((CurrencyId::try_from_symbol(symbol)?, amount)))
             .into_iter()
-            .collect::<Result<Vec<_>, Error>>()
-            .map_err(ServiceError::Abort)?;
+            .collect::<Result<Vec<_>, Error>>()?;
 
         // exit if auto-register uses faucet and faucet url not set
         if parsed_auto_register.iter().any(|(_, o)| o.is_none()) && self.config.faucet_url.is_none() {
             // TODO: validate before bitcoin / parachain connections
-            return Err(ServiceError::Abort(Error::FaucetUrlNotSet));
+            return Err(BackoffError::Permanent(Error::FaucetUrlNotSet));
         }
 
         let num_confirmations = match self.config.btc_confirmations {
             Some(x) => x,
-            None => self.btc_parachain.get_bitcoin_confirmations().await?,
+            None => self
+                .btc_parachain
+                .get_bitcoin_confirmations()
+                .await
+                .map_err(|err| BackoffError::Transient::<Error>(err.into()))?,
```

Comment:
I think before we automatically cast to `Retry`, is this no longer the case (for `Transient`)?

Reply:
The error returned by the `get_bitcoin_confirmations` method is `RuntimeError` and not `crate::Error`, hence I need to explicitly convert it. In the recent commit, I have just mapped it to `crate::Error` instead of `backoff:Error`. There is already `from` trait implemented in backoff::Error, check it out [here](https://docs.rs/backoff/latest/src/backoff/error.rs.html#110)